### PR TITLE
support rust native value

### PR DESCRIPTION
**Pull request by LeuisKen**

### CODE REVIEW

File: bridge/core/frame/module_manager.cc [hunk 1]
```diff
@@ -33,42 +33,56 @@ NativeValue* handleInvokeModuleTransientCallback(void* ptr,
...
-  }
+    if (exception_state.HasException()) {
+      context->HandleException(exception_state);
+      return nullptr;
+    }
 
-  return return_value;
+    return return_value;
+  } else if (auto* callback = DynamicTo<WebFNativeFunction>(callback_value.get())) {
+    if (errmsg != nullptr) {
```
Comment (andycall) [inline bridge/core/frame/module_manager.cc:76]: Missing return value impl?

**Review by andycall — APPROVED** [review]

File: bridge/core/frame/module_manager.cc (hunk 1 above)
Comment (andycall) [inline bridge/core/frame/module_manager.cc:76]: Missing return value impl?

**Review by andycall — CHANGES_REQUESTED** [review]

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+      values.push(val);
+    }
+    values
+  }
+}
+
+impl Drop for NativeValue {
+  fn drop(&mut self) {
+    if self.tag == NativeTag::TagString as i32 {
+      println!("Drop NativeValue string: {}", self.to_string());
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:210]: Missing impl for release string heap in rust

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+  pub fn to_string(&self) -> String {
+    let ptr = unsafe {
+      self.u.ptr as *mut SharedNativeString
+    };
+    let string_struct = unsafe { ptr.read() };
+    let slice = unsafe { std::slice::from_raw_parts(string_struct.string_, string_struct.length_.try_into().unwrap()) };
+    String::from_utf16_lossy(slice)
+  }
+
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:113]: Missing pub fn is_string(&self) -> bool {}

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    self.tag == NativeTag::TagNull as i32
+  }
+
+  pub fn new_float64(val: f64) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagFloat64 as i32;
+    value.u.float64 = val;
+    value.uint32 = 0;
+    value
+  }
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:132]: Missing pub fn is_float64(&self) -> bool {}

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+  pub fn new_int64(val: i64) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagInt as i32;
+    value.u.int64 = val;
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn to_int64(&self) -> i64 {
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:162]: Missing pub fn is_int(&self)

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+  pub fn new_bool(val: bool) -> Self {
+    let mut value = Self::new();
+    value.tag = NativeTag::TagBool as i32;
+    value.u.int64 = if val { 1 } else { 0 };
+    value.uint32 = 0;
+    value
+  }
+
+  pub fn to_bool(&self) -> bool {
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:148]: Missing pub fn is_bool(&self)

File: bridge/rusty_webf_sys/src/native_value.rs [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    let ptr = unsafe {
+      self.u.ptr as *mut NativeValue
+    };
+    for i in 0..self.uint32 {
+      let offset = i.try_into().unwrap();
+      let val = unsafe { ptr.add(offset).read() };
+      values.push(val);
+    }
+    values
+  }
```
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:204]: Missing pub fn is_list(&self)

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 2 above)
Comment (andycall) [inline bridge/rusty_webf_sys/src/native_value.rs:210]: careful about the malloc issue.

**Review by andycall — APPROVED** [review]

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 2 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:239]: updated.

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 3 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:120]: added

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 4 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:139]: added

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 5 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:181]: added

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 6 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:163]: added

File: bridge/rusty_webf_sys/src/native_value.rs (hunk 7 above)
Comment (LeuisKen) [reply bridge/rusty_webf_sys/src/native_value.rs:233]: added

**Pull request merged**